src/ui/mod.rs

```diff
@@ -0,0 +1,207 @@
+use indicatif::{MultiProgress, ProgressBar, ProgressStyle};
+use mockall::automock;
+use std::collections::HashMap;
+
+pub(crate) struct ProofUI {
+    main_progress: MultiProgress,
+    project_progress: Option<ProgressBar>,
+    seq_proof_progress: HashMap<String, ProgressBar>,
+    seq_proofstep_progress: HashMap<(String, String), ProgressBar>,
+    seq_oracle_progress: HashMap<(String, String, String), ProgressBar>,
+}
+
+#[automock]
```

Comment:
I think I have two thoughts about this:

1. I would like to keep all these dependencies dev-dependencies. It seems like that is not possible with double, not sure. I don't know how that would work with automock.
2. Keep the magic low. I honestly am not sure how this works. double replaces the struct definition and automock replaces the impl? Don't know.

Maybe we could define a trait for the UI (with the functions basically being the block below) and then implement that once for the normal UI type an then inside test-only code define a second implementation using the [mock! macro](https://docs.rs/mockall/latest/mockall/macro.mock.html). The verify functions et al could then be made generic over that trait. Is that too much work? We could move that to later.